### PR TITLE
webui: Add button to show verbose alert details in the alert, alert-log webui

### PR DESCRIPTION
**Pull request by theochita**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by theochita** [comment]
Based on Murrants comment in https://github.com/librenms/librenms/pull/12413

**Comment by theochita** [comment]
image

image

File: includes/html/common/alert-log.inc.php [hunk 1]
```diff
@@ -83,6 +83,7 @@
             <th data-column-id="hostname">Device</th>
```
Comment (Jellyfrog) [inline includes/html/common/alert-log.inc.php:83]: All files have the wrong permissons

**Review by Jellyfrog — CHANGES_REQUESTED** [review]

File: includes/html/common/alert-log.inc.php (hunk 1 above)
Comment (theochita) [reply includes/html/common/alert-log.inc.php:83]: should have fixed the permissions now.

**Comment by TheMysteriousX** [comment]
This change allows unprivileged users to view credentials used for device access - is this intended?

**Comment by theochita** [comment]
> This change allows unprivileged users to view credentials used for device access - is this intended?
When i copy a successful call to the ajax_form type "alert-details" and then logout and try again i get Unauthorized.

It is correct that it is not doing any user permission checks itself, normal users would be able to use it if they have the device permissions, giving them access to the community string so could be a potential source of leaks.

Making it global admin only is a possibility.

**Comment by theochita** [comment]
I made the use of this feature require admin permissions as per what @TheMysteriousX pointed out, better safe than sorry.

**Comment by Jellyfrog** [comment]
The tests fails

File: includes/html/common/alert-log.inc.php [hunk 2]
```diff
@@ -37,6 +37,10 @@
     'Warning' => 5,
 ];
 
+if (Auth::user()->hasGlobalAdmin()) {
+    $admin_verbose_details = '<th data-column-id="verbose_details" data-sortable="false">Details</th>';
```
Comment (murrant) [inline includes/html/common/alert-log.inc.php:41]: This only prevents the link from showing up, the details are still in the html below.  You'll need to add a check down there too.

**Review by theochita — COMMENTED** [review]
all tests passing.

**Comment by theochita** [comment]
Anything else i can do?

**Review by murrant — APPROVED** [review]
I'd prefer not to add legacy frontend code, but lgtm.

**Review by Jellyfrog — APPROVED** [review]

**Pull request merged**